react-native/react/actions/tracker.js

```diff
@@ -84,10 +86,40 @@ export function onFollowHelp (username: string): Action {
   }
 }
 
-export function onCloseFromActionBar (username: string): Action {
-  return {
-    type: Constants.onCloseFromActionBar,
-    payload: {username}
+function trackUser (username: string, state: {tracker: RootTrackerState}): Promise<boolean> {
```

Comment:
I don't mean to put the promo de there. I mean to have the result callback
just emit an action itself. Then the data goes into the store and if we
need to do something (like an error) we just dispatch another action.

On Thursday, December 3, 2015, Marco Munizaga <notifications@github.com>
wrote:

> In react-native/react/actions/tracker.js
> <https://github.com/keybase/client/pull/1470#discussion_r46640286>:
>
> > @@ -84,10 +86,40 @@ export function onFollowHelp (username: string): Action {
> >    }
> >  }
> >
> > -export function onCloseFromActionBar (username: string): Action {
> > -  return {
> > -    type: Constants.onCloseFromActionBar,
> > -    payload: {username}
> > +function trackUser (username: string, state: {tracker: RootTrackerState}): Promise<boolean> {
>
> I don't think we should be putting promises/functions in stores in general.
>
> I think it make sense that the action that triggers it cares about what
> happens to it.
>
> for example:
> say we block on closing until this returns successful, and if it errors we
> should dispatch a different action that will show some error info to the
> user. It would look something like this:
>
> trackUser.then(success => {
>   if (!success) throw failure...
>
>   dispatch( closeWindow )
> }).catch( () => {
>   dispatch( showError )
> })
>
> In this specific case the result doesn't matter and I'll remove the
> promise, but that may not always be true
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1470/files#r46640286>.
>